### ISSUE TITLE
[Inductor] [CPU] Huggingface model AllenaiLongformerBase performance regression > 10% on ww07.4

### ISSUE THREAD

**Issue by ESI-SYD** [pytorch/pytorch#95037]
### 🐛 Describe the bug

There is a performance regression on Huggingface model AllenaiLongformerBase  on ww07.4 [TorchInductor CPU Performance Dashboard](https://github.com/pytorch/pytorch/issues/93531#issuecomment-1434019577)  compare with ww07.2 as bellow:

|  ww07.4   |   |     |   |  ww07.2   |   |     |   | Comp  |     |   |
|  ----  | ----  |  ----  | ----  |  ----  | ----  |  ----  | ----  | ----  |  ----  | ----  |
| batch_size  | speedup | inductor  | eager | batch_size  | speedup | inductor  | eager | speedup ratio | eager ratio  | inductor ratio |
|1	|0.9901	|2.3819678	|2.358386319	|1	|1.1392	|2.0668015	|2.354500269	|0.87	|1	|0.87

Please find sw info from [TorchInductor CPU Performance Dashboard](https://github.com/pytorch/pytorch/issues/93531) 

### Versions

Repro:

```
python -m torch.backends.xeon.run_cpu --core_list 0 --ncores_per_instance 1 benchmarks/dynamo/huggingface.py --performance --float32 -dcpu -n50 --inductor  --no-skip --dashboard --only AllenaiLongformerBase  --cold_start_latency --batch_size 1 --threads 1
```

**Comment by leslie-fang-intel** [comment]
@blzheng please help to merge this issue to https://github.com/pytorch/pytorch/issues/95033

**Comment by leslie-fang-intel** [comment]
Sync with @blzheng offline, will try https://github.com/pytorch/pytorch/pull/116899 if it can help this issue.

**Comment by blzheng** [comment]
Guilty commit is https://github.com/pytorch/pytorch/commit/ceb0f1576b201adf8cafbde32a9ed7ba95ead025
The performance has improved, but it hasn't fully recovered. 
I conducted a preliminary analysis on T5Small, the profiling indicates that softmax kernel introduced a significant gap (totaling 70+ms). For example, in the profiling before regression, graph_0_kernel_cpp_8 occupied 6.793ms, whereas in the profiling from the latest code, graph_0_cpp_fused__softmax_7 costs 19.991ms. This indicates a gap of about 13ms for each softmax kernel. 
As shown in the generated code( [cpp_fused_softmax_7](https://gist.github.com/blzheng/21ec48334509aafc88af058c6607c83f#file-latest_output_code-py-L589-L745) , [kernel_cpp_8](https://gist.github.com/blzheng/6b199bfc4ab72937740b96bfbf89f901#file-before_regression_output_code-py-L611-L752) ) , cpp_fused__softmax_7 includes additional computations such as log. It could be caused by different handling methods for the inplace add before softmax op ([before](https://gist.github.com/blzheng/6b199bfc4ab72937740b96bfbf89f901#file-before_regression_fx_graph_readable-py-L124-L125) vs [after](https://gist.github.com/blzheng/21ec48334509aafc88af058c6607c83f#file-latest_fx_graph_readable-py-L171-L174))



Before regression:
![image](https://github.com/pytorch/pytorch/assets/69951214/8d012262-628f-4fb4-a79e-69b001e4a514)

latest codebase:
![image](https://github.com/pytorch/pytorch/assets/69951214/32075c36-f636-4ad5-8e5a-2a016b50567b)
